server/bin/pbench-server-put-shim.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3
+# -*- mode: python -*-
+
+# This script is used to send the tarballs that a version 002 client
+# submits for further processing. It forwards the tarballs and their MD5
+# sums to the remote server version 0.72 using the server's HTTP PUT method
+
+import os
+import sys
+import tempfile
+
+from pbench import PbenchConfig, BadConfig, get_pbench_logger
+from pbench.process_tb import ProcessTb
+from pbench.report import Report
+
+
+_NAME_ = "pbench-server-put-shim"
```

Comment:
I will look more into this

Reply:
While the hyphen-separated naming is awkward for direct unit testing, it'll also look weird to have some hyphenated names and other underscored names; I don't much like that inconsistency. Nor would I want to be changing all of the 0.69 commands to underscores at this point.

In the longer run (not in 0.69 of course) we want to move away from having "commands" for the server pipeline at all; it should be non-CLI microcomponents driven by a message service.

For unit testing, we really want to minimize the scope of the "shell" wrappers we use to launch things, and put all the testable logic in Python modules we can unit test directly. Yeah, ideally we want to test the shells as well, with a better framework than the current "gold" test system; but I wouldn't want to waste a lot of time worrying about that in 0.69.

Reply:
In the name of putting 0.69 to bed, I withdraw the comment.  (It just looked like we were adding a bunch of stuff that we didn't need because we were promulgating the flawed historical model.  However, in no case was I suggesting that we do anything with _existing_ hyphenated modules!)

And, it looks like Riya has largely gotten around the hyphen/unit-test problem by having the hyphenated module be just a wrapper which invokes guts which are in a separate module for which she's provided unit tests.